eventing/samples/github-events/README.md

```diff
@@ -0,0 +1,162 @@
+    can deploy the function into your cluster. **Ensure that the container image
+    value in `function.yaml` matches the container you built in the previous
+    step.** Apply the configuration using `kubectl`:
+
+    ```shell
+    kubectl apply -f eventing/samples/github-events/function.yaml
+    ```
+
+1.  Check that your service is running using:
+
+    ```shell
+    kubectl get services.serving.knative.dev -o "custom-columns=NAME:.metadata.name,READY:.status.conditions[2].status,REASON:.status.conditions[2].message"
+    NAME              READY     REASON
+    legit             True      <none>
+    ```
+
+1.  Create a [personal access token](https://github.com/settings/tokens) to 
+    GitHub repo that the GitHub source can use to register webhooks with the 
+    GitHub API. Also decide on a token that your code will use to authenticate
+    the incoming webhooks from GitHub (*accessToken*).
+    
+    The token can be named anything you find convenient. This sample requires
+    full `repo` control to be able update the title of the _Pull Request_.
+    The Source requires `admin:repo_hook`, this allows it to create webhooks
+    into repos that your account is allowed to do so. Copy and save this token;
+    GitHub will force you to generate it again if misplaced.
+
+    Here I named my token "EventingSample" and have selected the recommended
+    scopes: 
+
+    ![GitHub UI](personal_access_token.png "GitHub personal access token screenshot")
+
+    Update `eventing/samples/github-events/githubsecret.yaml` with those
+    values. If  your generated access token is `'asdfasfdsaf'` and you choose
+    your *secretToken* as `'personal_access_token_value'`, you'd modify
+    `eventing/samples/github-events/githubsecret.yaml` like so:
+
+    ```yaml
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: githubsecret
+    type: Opaque
+    stringData:
+      githubCredentials: >
+        {
+          "accessToken": "asdfasfdsaf",
+          "secretToken": "personal_access_token_value"
+        }
+    ```
+
+    Hint: you can makeup a random *accessToken* with:
+    
+    ```shell
+    head -c 8 /dev/urandom | base64
+    ```
+
+    Then, apply the githubsecret using `kubectl`:
+    
+    ```shell
+    kubectl apply -f eventing/samples/github-events/githubsecret.yaml
+    ```
+
+1.  Create the flow sending GitHub Events to the service:
+
+    ```shell
+    kubectl apply -f eventing/samples/github-events/flow.yaml
+    ```
+
+1.  Create a PR for the repo you configured the webhook for, and you'll see that
+    the Title will be modified with the suffix `(looks pretty legit)`
+
+
+## Understanding what happened
+
+<!--TODO:
+explain the resources and communication channels, as well as where the secret
+is used. In particular include a note to look at
+https://github.com/<owner>/<repo>/settings/hooks to see the webhook registered
+and then deleted.
+ -->
+
+## Cleaning up
+
+To clean up the function, `Flow`, auth, and secret:
```

Comment:
Do we need to clean up `auth.yaml`? I think it's needed to make the github source able to operate at all.

Reply:
Yes but I need help because I have not looked into what rbac arguments should look like